### PR TITLE
[api] Use proper :filename instead of :file in public_build route

### PR DESCRIPTION
**Pull request by ChrisBr**
Otherwise the filename constraint does not match and we end up
in rather random rails route matching.

Supersedes #2935 and fixes the failing tests.

cc @coolo

### CODE REVIEW

File: src/api/spec/routing/api_matcher_spec.rb [hunk 1]
```diff
@@ -17,14 +17,14 @@
     it { expect(get("/public/configuration?format=#{format}")).to route_to(controller: 'public', action: 'configuration_show', format: format) }
 
     it {
-      expect(get("/public/build/project/repository/arch/package/file.#{format}")).
+      expect(get("/public/build/project/repository/arch/package/filename?format=#{format}")).
```
Comment (bgeuken) [inline src/api/spec/routing/api_matcher_spec.rb]: That part could have stayed the same;-)

File: src/api/config/routes.rb [hunk 2]
```diff
@@ -637,7 +637,7 @@ def self.public_or_about_path?(request)
       get 'public/source/:project/:package/:filename' => :source_file, constraints: cons
       get 'public/distributions' => :distributions
       get 'public/binary_packages/:project/:package' => :binary_packages, constraints: cons
-      get 'public/build/:project(/:repository(/:arch(/:package(/:file))))' => 'public#build', constraints: cons, as: :public_build
+      get 'public/build/:project(/:repository(/:arch(/:package(/:filename))))' => 'public#build', constraints: cons, as: :public_build
```
Comment (bgeuken) [inline src/api/config/routes.rb]: That would be an API change, right?

Comment (bgeuken) [inline src/api/config/routes.rb]: Ah, nevermind. file already got replaced in the controller [4 years](54a74c760ab8e313f64b6d6d4efd0ea5ab047cdb) ago.

File: src/api/spec/routing/api_matcher_spec.rb (hunk 1 above)
Comment (ChrisBr) [inline src/api/spec/routing/api_matcher_spec.rb]: Yes but I wanted to have it consistent: arch -> arch, package -> package and then filename -> file?
And it was anyway necessary to change the format ...

**Comment by coolo** [comment]
cool. I knew something was fishy :)

**Pull request merged**